media/js/base/mozilla-image-helper.js

```diff
@@ -52,57 +47,24 @@ Mozilla.ImageHelper.initPlatformImages = function() {
             data_attribute += site.platform;
         }
 
-        if (is_high_res) {
-            suffix = '-high-res';
-        }
-        else {
-            suffix = '';
-        }
+        new_src = $img.data(data_attribute);
 
-        new_src = $img.data(data_attribute + suffix);
         if (!new_src) {
             // fall back to windows
-            new_src = $img.data('src-windows' + suffix);
+            data_attribute = 'src-windows';
+            new_src = $img.data(data_attribute);
         }
 
         this.src = new_src;
-        $img.attr('data-processed', 'true');
-        $img.addClass(site.platform);
-    });
-};
-
-// }}}
 
-// High Resolution Images
-// {{{ initHighResImages()
-
-Mozilla.ImageHelper.initHighResImages = function() {
-    $('img[data-src][data-high-res="true"][data-processed="false"]').each(function() {
-        var $img = $(this);
-        var src = $img.data('src');
-        if (Mozilla.ImageHelper.isHighDpi()) {
-            src = $img.data('high-res-src');
+        // if high res requested and path provided, set srcset attribute
+        if (is_high_res && $img.data(data_attribute + '-high-res')) {
```

Comment:
High res platform images still have the `data-src-{platform}-high-res` attributes, which are needed to correctly set/populate the `srcset` attribute.